packages/web3-eth-personal/test/integration/personal.test.ts

```diff
@@ -0,0 +1,125 @@
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { isHexStrict } from 'web3-validator';
+import { toChecksumAddress } from 'web3-utils';
+import { EthPersonal } from '../../src/index';
+import { importedAccount } from '../config/personal.test.config';
+import {
+	getSystemTestBackend,
+	getSystemTestAccounts,
+	getSystemTestProvider,
+	itIf,
+} from '../fixtures/system_test_utils';
+
+describe('personal integration tests', () => {
+	let ethPersonal: EthPersonal;
+	let accounts: string[];
+	beforeAll(() => {
+		const clientUrl = getSystemTestProvider();
+		ethPersonal = new EthPersonal(clientUrl);
+	});
+	beforeEach(async () => {
+		accounts = await getSystemTestAccounts();
+	});
+	it('new account', async () => {
+		const newAccount = await ethPersonal.newAccount('!@superpassword');
+		expect(isHexStrict(newAccount)).toBe(true);
+	});
+
+	itIf(getSystemTestBackend() === 'geth')('ecRecover', async () => {
+		// ganache does not support ecRecover
+		const signature = await ethPersonal.sign('0x2313', accounts[0], '');
+		const publicKey = await ethPersonal.ecRecover('0x2313', signature); // ecRecover is returning all lowercase
+		// eslint-disable-next-line jest/no-standalone-expect
+		expect(toChecksumAddress(publicKey)).toBe(toChecksumAddress(accounts[0]));
+	});
+
+	it('lock account', async () => {
+		const lockAccount = await ethPersonal.lockAccount(accounts[1]);
+		expect(lockAccount).toBe(true);
+	});
+
+	it('unlock account', async () => {
+		const key = accounts[0];
+		const unlockedAccount = await ethPersonal.unlockAccount(key, '', 100000);
+		expect(unlockedAccount).toBe(true);
+	});
+
+	// ganache does not support sign
+	itIf(getSystemTestBackend() === 'geth')('sign', async () => {
+		const key = accounts[0];
+		await ethPersonal.unlockAccount(key, '', 100000);
+		const signature = await ethPersonal.sign('0xdeadbeaf', accounts[0], '');
+		const address = await ethPersonal.ecRecover('0xdeadbeaf', signature);
+		// eslint-disable-next-line jest/no-standalone-expect
+		expect(accounts[0]).toBe(address);
+	});
+
+	it('getAccounts', async () => {
+		const accountList = await ethPersonal.getAccounts();
+		const accountsLength = accountList.length;
+		// create a new account
+		await ethPersonal.newAccount('cde');
+		const updatedAccountList = await ethPersonal.getAccounts();
+		expect(updatedAccountList).toHaveLength(accountsLength + 1);
```

Comment:
Instead of testing the length of `updatedAccountList`, I think it would make sense to test the expected value with something like:

```typescript
const newAccount = await ethPersonal.newAccount('cde');
expect(updateAccountList).toStrictEqual([newAccount]);
```

You can probably get rid of

```typescript
const accountList = await ethPersonal.getAccounts();
const accountsLength = accountList.length;
```

Since my suggested change would also be testing the length